lib/arsi/arel_tree_manager.rb

```diff
@@ -4,23 +4,25 @@ module Arsi
   module ArelTreeManager
     # This is from Arel::SelectManager which inherits from Arel::TreeManager.
     # We need where_sql on both Arel::UpdateManager and Arel::DeleteManager so we add it to the parent class.
-    if ::Arel::VERSION[0].to_i >= 6
-      def where_sql(engine = Arel::Table.engine)
-        return if @ctx.wheres.empty?
+    def where_sql(provided_engine = :none)
+      return if @ctx.wheres.empty?
 
-        viz = if ::Arel::VERSION[0].to_i > 6
-          ::Arel::Visitors::WhereSql.new(engine.connection.visitor, engine.connection)
+      selected_engine = provided_engine
+      if selected_engine == :none
+        # Arel 9 has no engine reader
+        selected_engine = if self.respond_to?(:engine)
```

Comment:
can do a constant assignment like `AREL_ENGINE = version > 9.0` or so to avoid the runtime check and make it easy to delete once we are on arel 9

Reply:
I will. I don't know how to make assumptions about what will change in different versions so I checked for the behavior instead.